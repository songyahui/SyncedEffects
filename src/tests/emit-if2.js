"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module A B C>
	<hh.parallel>
	  <hh.loop>
	    <hh.emit A ifApply=${function(){return this.B.nowval > 3}}/>
	    <hh.pause/>
	  </hh.loop>
	  <hh.loop>
	    <hh.if C>
	      <hh.emit B value=${4}/>
	      <hh.emit B value=${3}/>
	    </hh.if>
	    <hh.pause/>
	  </hh.loop>
	</hh.parallel>
      </hh.module>

const m = new hh.ReactiveMachine(prg);
m.debug_emitted_func = console.log

m.react()
m.react()
m.inputAndReact("C")
m.react()
m.inputAndReact("C")
m.inputAndReact("C")

