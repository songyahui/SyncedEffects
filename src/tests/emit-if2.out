[ 'B(3)' ]
[ 'B(3)' ]
[ 'A', 'B(4)', 'C' ]
[ 'B(3)' ]
[ 'A', 'B(4)', 'C' ]
[ 'A', 'B(4)', 'C' ]
